spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/MongoDbChannelMessageStore.java

```diff
@@ -96,12 +99,23 @@ public boolean isPriorityEnabled() {
 	@Override
 	public void afterPropertiesSet() {
 		super.afterPropertiesSet();
-		getMongoTemplate()
-				.indexOps(this.collectionName)
-				.ensureIndex(new Index(MessageDocumentFields.GROUP_ID, Sort.Direction.ASC)
-						.on(MessageDocumentFields.PRIORITY, Sort.Direction.DESC)
-						.on(MessageDocumentFields.LAST_MODIFIED_TIME, Sort.Direction.ASC)
-						.on(MessageDocumentFields.SEQUENCE, Sort.Direction.ASC));
+		if (this.createIndex) {
```

Comment:
Is this implementation, okay? The parent `AbstractConfigurableMongoDbMessageStore` still creates indexes. Should I check in the parent class or is it okay for the parent class to create indexes even if the createIndex is set to false?

Reply:
Ok. I missed the fact that super class does create indexes as well. Thank for the pointer !
Then we need to push this property to super class and then we need to have a protected `createIndexes()` method which can be overridden in impels for their own indexes.